compiler/rustc_trait_selection/src/traits/structural_match.rs

```diff
@@ -258,16 +248,16 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for Search<'a, 'tcx> {
             let ty = self.tcx().normalize_erasing_regions(ty::ParamEnv::empty(), field_ty);
             debug!("structural-match ADT: field_ty={:?}, ty={:?}", field_ty, ty);
 
-            if ty.visit_with(self) {
+            if ty.visit_with(self) == ControlFlow::BREAK {
```

Comment:
yeah, we probably want to store `self.found` in `ControlFlow::Break` here.

(also not part of this PR)